### PR TITLE
Fix H5Pset_evict_on_close failing with non MPI-IO use

### PR DESCRIPTION
**Pull request by glennsong09**
Adds in a check to see if the the MPIO FD is being used and if the MPI size is greater than one before failing.

### CODE REVIEW

File: src/H5Pfapl.c [hunk 1]
```diff
@@ -4869,8 +4871,15 @@ H5Pset_evict_on_close(hid_t fapl_id, hbool_t H5_ATTR_PARALLEL_UNUSED evict_on_cl
     if (H5P_set(plist, H5F_ACS_EVICT_ON_CLOSE_FLAG_NAME, &evict_on_close) < 0)
         HGOTO_ERROR(H5E_PLIST, H5E_CANTSET, FAIL, "can't set evict on close property");
 #else
-    HGOTO_ERROR(H5E_PLIST, H5E_UNSUPPORTED, FAIL,
-                "evict on close is currently not supported in parallel HDF5");
+    if (H5Pget_driver(fapl_id) == H5FD_MPIO) {
```
Comment (derobins) [inline src/H5Pfapl.c:4874]: You can't make public API calls in internal functions

File: src/H5Pfapl.c [hunk 2]
```diff
@@ -4852,6 +4852,8 @@ H5Pset_evict_on_close(hid_t fapl_id, hbool_t H5_ATTR_PARALLEL_UNUSED evict_on_cl
 {
     H5P_genplist_t *plist;               /* property list pointer */
     herr_t          ret_value = SUCCEED; /* return value */
+    MPI_Comm        comm;                /* MPI comm pointer */
+    int             mpi_size;            /* MPI size */
```
Comment (derobins) [inline src/H5Pfapl.c:4856]: These should be protected with H5_HAVE_PARALLEL ifdefs

File: src/H5Pfapl.c [hunk 3]
```diff
@@ -4869,8 +4871,15 @@ H5Pset_evict_on_close(hid_t fapl_id, hbool_t H5_ATTR_PARALLEL_UNUSED evict_on_cl
...
+    if (H5Pget_driver(fapl_id) == H5FD_MPIO) {
+        H5Pget_fapl_mpio(fapl_id, &comm, NULL);
+        MPI_Comm_size(comm, &mpi_size);
+        MPI_Comm_free(&comm);
+        if (mpi_size > 1) {
+            HGOTO_ERROR(H5E_PLIST, H5E_UNSUPPORTED, FAIL,
+                        "evict on close is currently not supported in parallel HDF5");
+        }
+    }
 #endif /* H5_HAVE_PARALLEL */
```
Comment (derobins) [inline src/H5Pfapl.c:4883]: If your intent is to allow setting this property when mpi_size is 1, you aren't setting it when the checks succeed.

**Review by derobins — CHANGES_REQUESTED** [review]
Needs changes (see comments - should be easy to fix)

**Comment by jhendersonHDF** [comment]
@glennsong09 and I discussed these changes and have a different approach to fixing the issue

**Comment by derobins** [comment]
> @glennsong09 and I discussed these changes and have a different approach to fixing the issue

Do we need to close this, then?

**Comment by jhendersonHDF** [comment]
Closing this PR in favor of changes coming in a different PR

**Pull request closed without merging**